tests/iter.rs

```diff
@@ -1,100 +1,87 @@
 #![allow(deprecated)]
 
-use ndarray::array;
-use numpy::{pyarray, NpyMultiIterBuilder, NpySingleIterBuilder, PyArray};
-use pyo3::{PyResult, Python};
-
-macro_rules! assert_approx_eq {
-    ($x: expr, $y: expr) => {
-        assert!(($x - $y) <= std::f64::EPSILON);
-    };
-}
+use numpy::{pyarray, NpyMultiIterBuilder, NpySingleIterBuilder};
+use pyo3::Python;
+
+// The order of iteration is not specified, so we should restrict ourselves
+// to tests that don't verify a given order.
 
 #[test]
-fn readonly_iter() -> PyResult<()> {
-    pyo3::Python::with_gil(|py| {
-        let data = array![[0.0, 1.0], [2.0, 3.0], [4.0, 5.0]];
-
-        let arr = PyArray::from_array(py, &data);
-        let iter = NpySingleIterBuilder::readonly(arr.readonly()).build()?;
-
-        // The order of iteration is not specified, so we should restrict ourselves
-        // to tests that don't verify a given order.
-        assert_approx_eq!(iter.sum::<f64>(), 15.0);
-        Ok(())
-    })
+fn readonly_iter() {
+    Python::with_gil(|py| {
+        let arr = pyarray![py, [0, 1], [2, 3], [4, 5]];
+
+        let iter = NpySingleIterBuilder::readonly(arr.readonly())
+            .build()
+            .unwrap();
+
+        assert_eq!(iter.sum::<i32>(), 15);
+    });
 }
 
 #[test]
-fn mutable_iter() -> PyResult<()> {
-    let data = array![[0.0, 1.0], [2.0, 3.0], [4.0, 5.0]];
-    pyo3::Python::with_gil(|py| {
-        let arr = PyArray::from_array(py, &data);
-        let iter = unsafe { NpySingleIterBuilder::readwrite(arr).build()? };
+fn mutable_iter() {
+    Python::with_gil(|py| {
+        let arr = pyarray![py, [0, 1], [2, 3], [4, 5]];
+
+        let iter = unsafe { NpySingleIterBuilder::readwrite(arr).build().unwrap() };
+
         for elem in iter {
-            *elem *= 2.0;
+            *elem *= 2;
```

Comment:
I prefer to note the decimal to clarify that `*elem` is a float, but I'm not sure about the community's preference or best practices.

Reply:
The type of `elem` was changed to `i32` by
```rust
let arr = pyarray![py, [0, 1], [2, 3], [4, 5]];
```

I changed all tests (where that does not affect their meaning) to work with integers as this avoids having to do tricks like `assert_approx_eq!` to check for equality against expectations. With integers, I can for example just use `assert_eq!` to compare the exact sums.